File: armstrong.java
import java.util.*;
public class armstrong{
    public static void main(String[] args) {
        System.out.println("Enter the number");
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int digits=0;
        while(n!=0){
            n/=10;
            digits++;
        }
        int sum=0;
        while(n!=0){
            int digit=n%10;
            sum+=Math.pow(digit,digits);
            n/=10;
        }
        if(sum==n){
            System.out.println("It is a armstrong ");
        }
        else{
            System.out.println("Not a");
        }
        

        
    }
}

File: armstrong.py
import sys


def is_armstrong(num):
    # Computes whether a number equals the sum of its digits raised to the digit count
    digits = list(map(int, str(abs(num))))
    power = len(digits)
    return num == sum(d ** power for d in digits)


def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    if is_armstrong(n):
        print("It is a armstrong ")
    else:
        print("Not a")


if __name__ == "__main__":
    main()
